### PR TITLE
protos: disambiguate `canonicalized_body`

### LINKED ISSUE #62: Ambiguity around `TransparencyLogEntry.canonicalized_body`
I noticed this while trying to get `sigstore-python`'s WIP bundle generation verifying against `sigstore-js`: here's what the current `canonicalized_body` docs say:

```
        // The canonicalized Rekor entry body, used for SET verification. This
        // is the same as the body returned by Rekor. It's included here for
        // cases where the client cannot deterministically reconstruct the
        // bundle from the other fields. Clients MUST verify that the signature
        // referenced in the canonicalized_body matches the signature provided
        // in the bundle content.
```

I interpreted this as the _body_ of the response returned by Rekor, without the surrounding `apiVersion`, `kind`, etc. envelope. My reasoning there is that that's the canonicalized body that goes directly into SET verification, without any transformation, while including the envelope means that the user has to remove it and then potentially re-canonicalize. 

`sigstore-js` interprets it the other way: they include and expect the entire enveloped form.

IMO either of these interpretations is reasonable, but there probably needs to be exactly one 🙂

### PR DESCRIPTION
**Pull request by woodruffw**
Closes #62.

Let me know if this is too verbose 😅 

Signed-off-by: William Woodruff <william@trailofbits.com>

### CODE REVIEW

**Comment by woodruffw** [comment]
cc @haydentherapper

**Review by asraa — APPROVED** [review]
Thank you for clarifying the client responsibility here! LGTM

Only curious if you want to include an example of one. But I actually think example full bundles should be written i a separate doc away from the codegen.

**Comment by woodruffw** [comment]
> Only curious if you want to include an example of one. But I actually think example full bundles should be written i a separate doc away from the codegen.

Yeah, I think an example would be beneficial, but agreed about putting it somewhere else 🙂

File: protos/sigstore_rekor.proto [hunk 1]
```diff
@@ -103,11 +103,18 @@ message TransparencyLogEntry {
...
         InclusionProof inclusion_proof = 6;
-        // The canonicalized Rekor entry body, used for SET verification. This
-        // is the same as the body returned by Rekor. It's included here for
-        // cases where the client cannot deterministically reconstruct the
-        // bundle from the other fields. Clients MUST verify that the signature
-        // referenced in the canonicalized_body matches the signature provided
-        // in the bundle content.
+        // The canonicalized transparency log entry, used to reconstruct
+        // the Signed Entry Timestamp (SET) during verification.
+        // The contents of this field as the same as the `body` field in
```
Comment (haydentherapper) [inline protos/sigstore_rekor.proto:108]:
```suggestion
        // The contents of this field are the same as the `body` field in
```

**Review by haydentherapper — APPROVED** [review]
Thanks!

**Review by haydentherapper — APPROVED** [review]

**Review by bdehamer — APPROVED** [review]

**Pull request merged**